Failed to compile client target due to missing libm linkage

## What?

When compiled with system openap-soft headers (v1.23.1) on Linux environment, I ran into following errors:

```
/usr/bin/ld: CMakeFiles/client.dir/client/entities.c.o: undefined reference to symbol 'atan2@@GLIBC_2.17'
/usr/bin/ld: /usr/lib64/libm.so.6: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/client.dir/build.make:2961: /home/tle/Work/Q2RTX/q2rtx] Error 1
make[1]: *** [CMakeFiles/Makefile2:533: src/CMakeFiles/client.dir/all] Error 2
```

## How to reproduce?

```
$  sudo dnf info openal-soft-devel
[sudo] password for tle:
Name         : openal-soft-devel
Version      : 1.23.1

$ gcc --version
gcc (GCC) 14.0.1 20240316 (Red Hat 14.0.1-0)

$ cmake --version
cmake version 3.28.2

$ cmake .. -DCMAKE_BUILD_TYPE=Release -DUSE_SYSTEM_OPENAL=true && make -j256
```

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -592,6 +592,9 @@ endif()
 
 IF(UNIX)
     TARGET_LINK_LIBRARIES(server dl rt m pthread)
+    IF(TARGET client)
+        TARGET_LINK_LIBRARIES(client m)
+    ENDIF()
 ENDIF()
 
 IF(TARGET client)

